tests/e2e/democracy_test.go

```diff
@@ -0,0 +1,257 @@
+	currentProviderFeeAccountBalance := sdk.NewDecFromInt(bankKeeper.GetBalance(s.consumerCtx(), providerRedistributeAccount.GetAddress(), bondDenom).Amount)
+	currentCommunityPoolBalance := distrKeeper.GetFeePoolCommunityCoins(s.consumerCtx()).AmountOf(bondDenom)
+	for key := range currentRepresentativesRewards {
+		representativeAddr, _ := sdk.ValAddressFromBech32(key)
+		representativeReward := distrKeeper.GetValidatorOutstandingRewards(s.consumerCtx(), representativeAddr).Rewards.AmountOf(bondDenom)
+		currentRepresentativesRewards[key] = representativeReward
+	}
+
+	s.consumerChain.NextBlock()
+
+	nextDistrModuleAccountBalance := sdk.NewDecFromInt(bankKeeper.GetBalance(s.consumerCtx(), distrModuleAccount.GetAddress(), bondDenom).Amount)
+	nextProviderFeeAccountBalance := sdk.NewDecFromInt(bankKeeper.GetBalance(s.consumerCtx(), providerRedistributeAccount.GetAddress(), bondDenom).Amount)
+	nextCommunityPoolBalance := distrKeeper.GetFeePoolCommunityCoins(s.consumerCtx()).AmountOf(bondDenom)
+	for key := range nextRepresentativesRewards {
+		representativeAddr, _ := sdk.ValAddressFromBech32(key)
+		representativeReward := distrKeeper.GetValidatorOutstandingRewards(s.consumerCtx(), representativeAddr).Rewards.AmountOf(bondDenom)
+		nextRepresentativesRewards[key] = representativeReward
+	}
+
+	distrModuleDifference := nextDistrModuleAccountBalance.Sub(currentDistrModuleAccountBalance)
+	providerDifference := nextProviderFeeAccountBalance.Sub(currentProviderFeeAccountBalance)
+	communityPoolDifference := nextCommunityPoolBalance.Sub(currentCommunityPoolBalance)
+	representativeDifference := map[string]sdk.Dec{}
+	consumerRedistributeDifference := communityPoolDifference
+
+	for key, currentReward := range currentRepresentativesRewards {
+		representativeDifference[key] = nextRepresentativesRewards[key].Sub(currentReward)
+		consumerRedistributeDifference = consumerRedistributeDifference.Add(representativeDifference[key])
+	}
+
+	s.Require().Equal(distrModuleDifference, consumerRedistributeDifference)
+	s.Require().Equal(communityPoolDifference.Quo(consumerRedistributeDifference), distrKeeper.GetCommunityTax(s.consumerCtx()))
+	s.Require().InEpsilon(distrModuleDifference.Quo(providerDifference.Add(distrModuleDifference)).MustFloat64(), consumerFraction.MustFloat64(), float64(0.0001))
+	s.Require().InEpsilon(providerDifference.Quo(providerDifference.Add(distrModuleDifference)).MustFloat64(), sdk.NewDec(1).Sub(consumerFraction).MustFloat64(), float64(0.0001))
+
+	totalRepresentativePower := stakingKeeper.GetValidatorSet().TotalBondedTokens(s.consumerCtx())
+
+	for key, representativeTokens := range representativesTokens {
+		powerFraction := sdk.NewDecFromInt(representativeTokens).QuoTruncate(sdk.NewDecFromInt(totalRepresentativePower))
+		s.Require().Equal(powerFraction, representativeDifference[key].Quo(consumerRedistributeDifference.Sub(communityPoolDifference)))
```

Comment:
Comment: "check that each representative has gotten the correct amount of rewards"